Documentation/ApiOverview/Events/Events/Linkvalidator/BeforeRecordIsAnalyzedEvent.rst

```diff
@@ -10,6 +10,144 @@ BeforeRecordIsAnalyzedEvent
+        ) {
+            $this->brokenLinkRepository = $brokenLinkRepository;
+            $this->softReferenceParserFactory = $softReferenceParserFactory;
+        }
+    }
+
+In this example we only check the field :sql:`bodytext` of new tables:
+
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php
+
+    public function handleEvent(BeforeRecordIsAnalyzedEvent $event): void
+    {
+        $table = $event->getTableName();
+        $forbiddenDomain = 'example.org';
+        if ($table == 'tx_news_domain_model_news') {
+            $results = $event->getResults();
+            $newsItem = $event->getRecord();
+            // a quick check if the forbidden domain is contained at all
+            if (strpos((string) $newsItem['bodytext'], $forbiddenDomain) !== false) {
+                $field = 'bodytext';
+                $conf = $GLOBALS['TCA']['tx_news_domain_model_news']['columns']['bodytext']['config'];
+                // parse the field
+                foreach ($this->softReferenceParserFactory->getParsersBySoftRefParserList($conf['softref'], ['subst']) as $softReferenceParser) {
+                    $parserResult = $softReferenceParser->parse($table, $field, $newsItem['uid'], $newsItem['bodytext']);
+                    if (!$parserResult->hasMatched()) {
+                        continue;
+                    }
+                    // act on each link found
+                    foreach ($parserResult->getMatchedElements() as $matchedElement) {
+                        // if it contains the forbidden domain, subdomains would also be allerted here
+                        if ($matchedElement['subst'] && $matchedElement['subst']['tokenValue']
+                            && strpos($matchedElement['subst']['tokenValue'] . '', $forbiddenDomain) !== false) {
+                            // create an entry for the table tx_linkvalidator_link
+                            $link = [
+                                'record_uid' => $newsItem['uid'],
+                                'record_pid' => $newsItem['pid'],
+                                'language' => $newsItem['sys_language_uid'],
+                                'field' => $field,
+                                'table_name' => $table,
+                                'url' => $matchedElement['subst']['tokenValue'],
+                                'last_check' => time(),
+                                'link_type' => 'external',
+                            ];
+                            // Insert it into the repository with
+                            // A meaningfull exception text and a unique error number
+                            $this->brokenLinkRepository->addBrokenLink($link, false, [
+                                'errorType' => 'exception',
+                                'exception' => 'Do not link externally to ' . $forbiddenDomain,
+                                'errno' => 42]);
```

Comment:
What is this errno 42? Is it a fixed value somewhere defined or just a random value for this case? Would prefer a class constant with a speaking name to make it clear.